src/tests/tracing/eventpipe/userevents/userevents.cs

```diff
@@ -0,0 +1,120 @@
+
+        public static int TestEntryPoint()
+        {
+            string appBaseDir = AppContext.BaseDirectory;
+            string recordTracePath = Path.Combine(appBaseDir, "record-trace");
+            string scriptFilePath = Path.Combine(appBaseDir, "dotnet-common.script");
+            string traceFilePath = Path.Combine(appBaseDir, trace);
+
+            if (!File.Exists(recordTracePath) || !File.Exists(scriptFilePath))
+            {
+                Console.WriteLine("record-trace or dotnet-common.script not found. Test cannot run.");
+                return -1;
+            }
+
+            ProcessStartInfo traceeStartInfo = new();
+            traceeStartInfo.FileName = Process.GetCurrentProcess().MainModule.FileName;
+            traceeStartInfo.Arguments = $"{typeof(UserEventsTest).Assembly.Location} tracee";
+            traceeStartInfo.WorkingDirectory = appBaseDir;
+
+            ProcessStartInfo recordTraceStartInfo = new();
+            recordTraceStartInfo.FileName = recordTracePath;
+            recordTraceStartInfo.Arguments = $"--script-file {scriptFilePath}";
+            recordTraceStartInfo.WorkingDirectory = appBaseDir;
+            recordTraceStartInfo.RedirectStandardOutput = true;
+            recordTraceStartInfo.RedirectStandardError = true;
+
+            using Process traceeProcess = Process.Start(traceeStartInfo);
+            using Process recordTraceProcess = Process.Start(recordTraceStartInfo);
+            recordTraceProcess.OutputDataReceived += (_, args) => Console.WriteLine($"[record-trace] {args.Data}");
+            recordTraceProcess.BeginOutputReadLine();
+            recordTraceProcess.ErrorDataReceived += (_, args) => Console.Error.WriteLine($"[record-trace] {args.Data}");
+            recordTraceProcess.BeginErrorReadLine();
+
+            if (!traceeProcess.HasExited && !traceeProcess.WaitForExit(15000))
+            {
+                traceeProcess.Kill();
+            }
+
+            // Until record-trace supports duration, the only way to stop it is to send SIGINT (ctrl+c)
+            kill(recordTraceProcess.Id, SIGINT);
+            if (!recordTraceProcess.HasExited && !recordTraceProcess.WaitForExit(20000))
+            {
+                // record-trace needs to stop gracefully to generate the trace file
+                recordTraceProcess.Kill();
+            }
+
+            if (!File.Exists(traceFilePath))
+            {
+                Console.Error.WriteLine($"Expected trace file not found at `{traceFilePath}`");
+                return -1;
+            }
+
+            if (!ValidateTraceeEvents(traceFilePath))
+            {
+                Console.Error.WriteLine($"Trace file `{traceFilePath}` does not contain expected events.");
+                return -1;
+            }
+
+            return 100;
+        }
+
+        private static bool ValidateTraceeEvents(string traceFilePath)
+        {
+            string etlxPath = TraceLog.CreateFromEventPipeDataFile(traceFilePath);
```

Comment:
You can parse the .nettrace file directly using EventPipeEventSource in TraceEvent. This avoids creating a 2nd file that the test also needs to clean up.

Reply:
For some reason when I tried it last, it didn't work (pilot error), switched to using EventPipeEventSource. Right now the events are "unknown" with id. Maybe its cause I'm using the Dynamic parser? I'll look into TraceEvent more closely

Reply:
it turned out it was just that AllocationSampled event is too new that PerfView doesn't know about it, Using an EventPipeEventSource by default now.